webapp-django/crashstats/base/tests/testbase.py

```diff
@@ -35,15 +35,22 @@ def setUp(self):
         # These are all the classes who have an implementation
         # (e.g. a class that belongs to socorro.external.something.something)
         # They all need to be mocked.
+        self._mockeries = {}
 
         for klass in classes_with_implementation:
+            self._mockeries[klass] = klass.implementation
+
             klass.implementation = mock.MagicMock()
             # The class name is used by the internal caching that guards
             # for repeated calls with the same parameter.
             # If we don't do this the cache key will always have the
             # string 'MagicMock' in it no matter which class it came from.
             klass.implementation().__class__.__name__ = klass.__name__
 
+    def tearDown(self):
+        for klass in classes_with_implementation:
+            klass.implementation = self._mockeries[klass]
+
```

Comment:
Previously, the `DjangoTestCase` would mock everything in `setUp` and then ... well, that was it. So the problem is that all those things stay mocked and if you run tests in a different order and one of them didn't expect them to be mocked, then it might fail or it might pass. I had one test that when run on its own worked fine, but when run with other tests, failed miserably with weird errors.

I fixed this by adding a `tearDown` that unmocks things. I suspect there are easier ways to do this, but I was trying hard to touch as little as I could in order to get my stuff working.

Reply:
I have a hell of a time using pytest fixtures with `unittest.TestCase` derived classes. I have no idea how it'd work with subclassing and method overriding. Otherwise, I like idea a lot!